randovania/game_description/integrity_check.py

```diff
@@ -177,14 +187,19 @@ def find_invalid_strongly_connected_components(game: GameDescription) -> Iterato
         if node not in graph:
             continue
 
-        for other, req in game.region_list.potential_nodes_from(node, context):
-            if other not in graph:
-                continue
+        try:
+            for other, req in game.region_list.potential_nodes_from(node, context):
+                if other not in graph:
+                    continue
 
-            if req != Requirement.impossible():
-                graph.add_edge(node, other)
+                if req != Requirement.impossible():
+                    graph.add_edge(node, other)
+
+        except KeyError:
+            # Broken docks
```

Comment:
could you elaborate a bit more on this comment?

Reply:
This PR is in need of a rebase because there's some things in this diff that are already in main 🤔 

I added that one in case the database has a dock to somewhere that doesn't exist. It'd cause the integrity check to raise which is bad.

Reply:
Can't even remember that part. I think that is from Dark and only here because it was based on #4499 .
Need to rebase this branch and resolve conflicts etc. That it has a more reviewable state ;)